_posts/2019-08-12-how-to-configure-production-grade-aws-account-structure.adoc

```diff
@@ -1223,9 +1381,11 @@ Finally, deploy the `account-baseline` module by running `terragrunt apply`:
 [source,bash]
 ----
 cd infrastructure-live/root/_global/account-baseline
-terragrunt apply
+terragrunt apply -parallelism=4
 ----
 
+IMPORTANT: We **strongly** recommend setting Terraform parallelism to a low value (e.g., `-parallelism=4`) with the `account-baseline-xxx` modules. This is because these modules deploy multi-region resources (e.g., GuardDuty, AWS Config, etc), and for each region, Terraform spins up a separate process, so if you don't limit the parallelism, it may peg all your CPU cores and lead to network connectivity errors.
```

Comment:
May also want to use `ulimit -n` to adjust the number of allowed open file descriptors.

Reply:
Could you provide a bit more context on how `ulimit` helps here?

Reply:
I guess it's really an adjacent problem, but I've consistently encountered errors like:

`Error: pipe: too many open files`

when running TF on large state files. The concern is that users may encounter the same issue, especially as their environments/state files grow.

Reply:
Weird, I've seen TLS / connection issues, but never too many open files issues...

Reply:
Just saw this when running `destroy` on account-baseline-app. My ulimit was 256. `ulimit -n 1024` fixes the issue.

Reply:
https://github.com/gruntwork-io/gruntwork-io.github.io/pull/352